### ISSUE TITLE
Remove all handling of `index.mapper.dynamic`

### ISSUE THREAD

**Issue by peternied** [opensearch-project/OpenSearch#12416]
### Description
`index.mapper.dynamic`'s deprecation predates OpenSearch.  We should remove any traces of this legacy setting from the system except to warn that the setting won't be actioned on.

The `INDEX_MAPPER_DYNAMIC_SETTING` should be ignored, it won't be, `org.opensearch.action.support.AutoCreateIndex`:

```
    public AutoCreateIndex(
        Settings settings,
        ClusterSettings clusterSettings,
        IndexNameExpressionResolver resolver,
        SystemIndices systemIndices
    ) {
        this.resolver = resolver;
        dynamicMappingDisabled = !MapperService.INDEX_MAPPER_DYNAMIC_SETTING.get(settings);
        this.systemIndices = systemIndices;
        this.autoCreate = AUTO_CREATE_INDEX_SETTING.get(settings);
        clusterSettings.addSettingsUpdateConsumer(AUTO_CREATE_INDEX_SETTING, this::setAutoCreate);
    }
```

- _Originally posted by @reta in https://github.com/opensearch-project/OpenSearch/issues/11193#issuecomment-1956830794_

### Expected Result
Remove this legacy settings usage expect for purposes of warning about its removal and non-functionality.

**Comment by peternied** [comment]
[Triage - attendees [1](https://github.com/peternied) [2](https://github.com/dbwiddis) [3](https://github.com/lukas-vlcek) [4](https://github.com/kotwanikunal) [5](https://github.com/linuxpi)]
Thanks for creating this issue

**Comment by reta** [comment]
> @peternied We need to fix this ASAP now that #11193 allows for indices to be created with this deprecated setting present, right?

So we need to check the `indexVersionCreated` and the version is on or after V_2_0_0, we should reject index creation, the same we have to do for 2.x backport but lower the version to ~1.1.0 since we do not expect any indices created by OpenSearch to have this setting).

**Comment by peternied** [comment]
> We need to fix this ASAP

This change was merged into 2.x, so it needs to target 2.13.  I'll work on this soon and want to be clear no 2.12.1 change is needed.
